submissions/zalevskyi/memory-pair-game/app.js

```diff
@@ -0,0 +1,112 @@
+class Card {
+    constructor(image, index) {
+        //image - object with file name and alt text of the image
+        //index - card index in current deal of the deck
+        this.image_file = image.file
+        this.div = document.createElement('div')
+        this.div.id = `card${index}`
+        this.div.classList.add('card')
+        this.div.innerHTML = [`<div class='back' id='back${index}'></div>`,
+                              `<div class='face'>`,
+                                `<img class='card_image' src='./images/${image.file}' alt='${image.alt} image'>`,
+                              `</div>`].join('')
+    }
+    open() {this.div.classList.add('opened')}
+    close() {this.div.classList.remove('opened')}
+    hide() {this.div.classList.add('hidden')}
+    static match(card1, card2) {
+        if (card1.image_file === card2.image_file) return true
+        else return false
+    }
+}
+//Card deck for memory pair matching
+class Deck {
+    constructor(imagesArray, cardsQuantityInDeal, showResult) {
+        //imagesArray - all available images (file name and alt text) to select from for a deal
+        //showResult - callback function to call on full deck match
+        this.imagesArray = imagesArray
+        this.cardsQuantityInDeal = cardsQuantityInDeal
+        this.showResult = showResult
+        this.currentDealCards = []
+        this.opened = []
+        this.matched_quantity = 0
+        this.flip_count = 0
+        this.shuffle()
+    }
+    shuffle() {
+        let currentDealImages = shuffleArray(this.imagesArray).slice(0, this.cardsQuantityInDeal / 2)
+        currentDealImages = shuffleArray(currentDealImages.concat(currentDealImages))
+        this.currentDealCards = currentDealImages.map((image, index) => new Card(image, index))
+        this.opened = []
+        this.matched_quantity = 0
+        this.flip_count = 0
+    }
+    placeOnBoard(board) {
+        //board is div element for appending child card elements
+        board.innerHTML = ''
+        this.currentDealCards.forEach(card => board.appendChild(card.div))
+    }
+    openCard(cardId) {
+        this.flip_count++
+        if (this.opened.length==2) this._closeOpened()
+        this.currentDealCards[cardId].open()
+        this.opened.push(this.currentDealCards[cardId])
+        this._checkOpenedMatch()
+    }
+    _checkOpenedMatch() {
+        if (this.opened.length==2 && Card.match(this.opened[0], this.opened[1])) {
```

Comment:
will be more readable with destructuring
`const [firstCard, secondCard] = this.opened;`

use strict equality operator